sw/device/mask_rom/mask_rom.c

```diff
@@ -3,13 +3,43 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "sw/device/mask_rom/mask_rom.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
 #include "sw/device/lib/base/stdasm.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/rom_exts/rom_ext_manifest_parser.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+// This is the expected ROM_EXT Manifest Identifier
+static const uint32_t kRomExtIdentifierExpected = 0x4552544F;
+
+typedef void(boot_fn)(void);
 
 void mask_rom_boot(void) {
-  asm volatile("unimp");
-  __builtin_unreachable();
+  rom_ext_manifest_t rom_ext = rom_ext_get_parameters(kRomExtManifestSlotA);
+
+  // Check we have a valid ROM_EXT
+  if (rom_ext_get_identifier(rom_ext) == kRomExtIdentifierExpected) {
+    // Set mtvec for ROM_EXT.
+    uintptr_t interrupt_vector = rom_ext_get_interrupt_vector(rom_ext);
```

Comment:
We have discussed this offline, regarding how this could be an issue with PMP (cyclic exceptions). However, this is just a note and not a subject to decision in this PR (should not block).

Reply:
Maybe is worth adding a `TODO` with a summary of my above comment?

Reply:
I've added something in `manifest.md` around where you had queries, it's marked as an open question.

Reply:
Great, thanks.